service/storage/storage_test.go

```diff
@@ -783,16 +783,164 @@ func TestFileDelete(t *testing.T) {
 	}
 }
 
+func TestFileListVersions(t *testing.T) {
+	testCases := []struct {
+		description    string
+		createdAtSince *strfmt.DateTime
+		createdAtUntil *strfmt.DateTime
+		calls          func(*mock.MockStorage) []*gomock.Call
+		expected       []*models.FileDescriptor
+		errorExpected  bool
+		exactError     error
+	}{
+		{
+			"BucketExsits fails",
+			nil,
+			nil,
+			func(s *mock.MockStorage) []*gomock.Call {
+				return []*gomock.Call{
+					s.EXPECT().BucketExists(gomock.Any(), "BUCKET").Return(false, fmt.Errorf("Error")),
+				}
+			},
+			nil,
+			withErrors,
+			nil,
+		},
+		{
+			"List fails",
+			nil,
+			nil,
+			func(s *mock.MockStorage) []*gomock.Call {
+				return []*gomock.Call{
+					s.EXPECT().BucketExists(gomock.Any(), "BUCKET").Return(true, nil),
+					s.EXPECT().List(gomock.Any(), "BUCKET", "File1").Return(nil, fmt.Errorf("Error")),
```

Comment:
invalid operation: s (variable of type *invalid type) has no field or method EXPECT